app/services/alerts/generate_and_save_alerts_and_benchmarks.rb

```diff
@@ -0,0 +1,81 @@
+module Alerts
+  class GenerateAndSaveAlertsAndBenchmarks
+    def initialize(school:, aggregate_school: nil, benchmark_result_generation_run: nil, framework_adapter: FrameworkAdapter)
+      @school = school
+      @aggregate_school = aggregate_school || AggregateSchoolService.new(school).aggregate_school
+      @benchmark_result_generation_run = benchmark_result_generation_run || BenchmarkResultGenerationRun.create!
+      @framework_adapter = framework_adapter
+    end
+
+    def perform(asof_date = Time.zone.today)
+      ActiveRecord::Base.transaction do
+        @alert_generation_run = AlertGenerationRun.create!(school: @school)
+        @benchmark_result_school_generation_run = BenchmarkResultSchoolGenerationRun.create!(school: @school, benchmark_result_generation_run: @benchmark_result_generation_run)
+
+        relevant_alert_types.each do |alert_type|
+          if alert_type.benchmark == true
+            service = GenerateAlertTypeRunResult.new(school: @school, aggregate_school: @aggregate_school, alert_type: alert_type, use_max_meter_date_if_less_than_asof_date: true)
+            service.benchmark_dates(asof_date).each do |benchmark_date|
```

Comment:
What we need to check is whether the benchmark dates that are being used are different to whatever the default is when `perform` is called without a date. This will require checking the various alert type classes to see if there's any major differences.